library/Icingadb/View/DowntimeRenderer.php

```diff
@@ -0,0 +1,255 @@
+use ipl\Html\Attributes;
+use ipl\Html\Html;
+use ipl\Html\HtmlDocument;
+use ipl\Html\HtmlElement;
+use ipl\Html\TemplateString;
+use ipl\Html\Text;
+use ipl\I18n\Translation;
+use ipl\Web\Common\ItemRenderer;
+use ipl\Web\Widget\Icon;
+use ipl\Web\Widget\Link;
+
+/** @implements ItemRenderer<Downtime> */
+class DowntimeRenderer implements ItemRenderer
+{
+    use Translation;
+    use TicketLinks;
+    use HostLink;
+    use ServiceLink;
+
+    /** @var int Current Time */
+    protected $currentTime;
+
+    /** @var int Duration */
+    protected $duration;
+
+    /** @var int Downtime end time */
+    protected $endTime;
+
+    /** @var bool Whether the downtime is active */
+    protected $isActive;
+
+    /** @var int Downtime start time */
+    protected $startTime;
+
+    /** @var bool Whether the state has been loaded */
+    protected $stateLoaded = false;
+
+    /**
+     * @var bool Whether the item is being rendered in the detail view of the associated object (host/service)
+     *
+     * When true:
+     *
+     * - Creation of the link of the associated object (host/service) is omitted from the title
+     * - The ticket link will be created
+     */
+    protected $isDetailView = false;
+
+    /**
+     * Set whether the item is being rendered in the detail view of the associated object (host/service)
+     *
+     * @param bool $state
+     *
+     * @return $this
+     */
+    public function setIsDetailView(bool $state = true): self
+    {
+        $this->isDetailView = $state;
+
+        return $this;
+    }
+
+    /**
+     * Load the state of the downtime
+     *
+     * @param Downtime $item
+     *
+     * @return void
+     */
+    protected function loadState(Downtime $item): void
+    {
+        if ($this->stateLoaded) {
+            return;
+        }
+
+        if (
+            isset($item->start_time, $item->end_time)
+            && $item->is_flexible
+            && $item->is_in_effect
+        ) {
+            $this->startTime = $item->start_time->getTimestamp();
+            $this->endTime = $item->end_time->getTimestamp();
+        } else {
+            $this->startTime = $item->scheduled_start_time->getTimestamp();
+            $this->endTime = $item->scheduled_end_time->getTimestamp();
+        }
+
+        $this->currentTime = time();
+
+        $this->isActive = $item->is_in_effect
+            || ($item->is_flexible && $item->scheduled_start_time->getTimestamp() <= $this->currentTime);
+
+        $until = ($this->isActive ? $this->endTime : $this->startTime) - $this->currentTime;
+        $this->duration = explode(' ', DateFormatter::formatDuration(
+            $until <= 3600 ? $until : $until + (3600 - ((int) $until % 3600))
+        ), 2)[0];
+
+        $this->stateLoaded = true;
+    }
+
+    public function assembleAttributes($item, Attributes $attributes, string $layout): void
+    {
+        $attributes->add(new Attributes(['class' => ['downtime', $item->is_in_effect ? 'in-effect' : '']]));
```

Comment:
In minimal mode the first item has no top padding if it is *in effect*. This is already the case in main, but we should do better here by not adding this class if the layout is *minimal*.